pkg/daemon/update.go

```diff
@@ -1555,13 +1557,33 @@ func (dn *Daemon) writeFiles(files []ign3types.File) error {
 		}
 
 		// To allow writing of "empty" files we'll allow source to be nil
-		contents := &dataurl.DataURL{}
+		var contentsBytes []byte
 		if file.Contents.Source != nil {
 			var err error
-			contents, err = dataurl.DecodeString(*file.Contents.Source)
+			contents, err := dataurl.DecodeString(*file.Contents.Source)
 			if err != nil {
 				return err
 			}
+			if file.Contents.Compression != nil {
+				switch *file.Contents.Compression {
+				case "":
+					contentsBytes = contents.Data
+				case "gzip":
+					reader, err := gzip.NewReader(bytes.NewReader(contents.Data))
+					defer reader.Close()
```

Comment:
`reader` is `nil` if `err` isn't, so this should be after the error check.

Reply:
Fixed thanks!